File: spring-integration/pollable-channels/priority-queue-channel/src/main/java/com/ddubson/integration/PrinterGateway.java
package com.ddubson.integration;

import org.springframework.integration.annotation.Gateway;
import org.springframework.integration.annotation.MessagingGateway;
import org.springframework.messaging.Message;

import java.util.concurrent.Future;

@MessagingGateway
public interface PrinterGateway {
    @Gateway(requestChannel = "inputChannel")
    Future<Message<String>> print(Message<?> message);
}

File: spring-integration/subscribable-channels/messaging-bridge/src/main/kotlin/com/ddubson/integration/PrinterGateway.kt
package com.ddubson.integration

import org.springframework.integration.annotation.Gateway
import org.springframework.integration.annotation.MessagingGateway
import org.springframework.messaging.Message
import java.util.concurrent.Future

@MessagingGateway
interface PrinterGateway {
    @Gateway(requestChannel = "bridgedPollableChannel")
    fun print(message: Message<*>): Future<Message<*>>
}